actionview/lib/action_view/rendering.rb

```diff
@@ -92,12 +92,16 @@ def rendered_format
       # Find and render a template based on the options given.
       # :api: private
       def _render_template(options) #:nodoc:
-        variant = options[:variant]
+        variant = options.delete(:variant)
+        assigns = options.delete(:assigns)
+        context = view_context
+
+        context.assign assigns if assigns
```

Comment:
What if we change `view_context` to take assigns parameter with default to `view_assigns` and we could build the object with the correct assigns instead of changing them?